lib/screens/registration_screen.dart

```diff
@@ -0,0 +1,179 @@
+              ),
+              Center(
+                child: ElevatedButton(
+                    onPressed: () async {
+                      final form = formkey.currentState!;
+                      if (form.validate()) {
+                        final email_ = email.text;
+                        final password_ = password.text;
+                      }
+
+                      await Firebase.initializeApp(
+                        options: DefaultFirebaseOptions.currentPlatform,
+                      );
+                      final email_ = email.text;
+                      final password_ = password.text;
+                      try {
+                        final UserCredential = await FirebaseAuth.instance
+                            .createUserWithEmailAndPassword(
+                                email: email_, password: password_);
+
+                        print(UserCredential);
+                      } on FirebaseAuthException catch (e) {
+                        if (e.code == 'weak-password') {
+                          print(e.runtimeType);
+                          print('Weak passwowrd');
+                          // ignore: use_build_context_synchronously
+                          return showDialog(
+                              context: context,
+                              builder: ((context) {
+                                return Container(
+                                  child: AlertDialog(
+                                    title: const Text(
+                                        'You entered weak password. Try some hard one !'),
+                                    actions: [
+                                      TextButton(
+                                        onPressed: () {
+                                          Navigator.of(context)
+                                              .pushNamedAndRemoveUntil(
+                                                  registerRoute,
+                                                  (route) => false);
+                                        },
+                                        child: const Text('Ok'),
+                                      )
+                                    ],
+                                  ),
+                                );
+                              }));
+                        } else if (e.code == 'email-already-in-use') {
+                          print(
+                              'you have already an account, You can simply login !');
+                          // ignore: use_build_context_synchronously
+                          return showDialog(
+                              context: context,
+                              builder: ((context) {
+                                return Container(
+                                  child: AlertDialog(
+                                    title: const Text(
+                                        '''You are already an user of our App.
```

Comment:
read on internet about RichText and use rich text to show text and textbutton in same line